ReactiveUI.XamForms/XamForms/ActivationForViewFetcher.cs

```diff
@@ -32,17 +31,61 @@ public IObservable<bool> GetActivationForView(IActivatable view)
                     Observable.FromEventPattern<EventHandler, EventArgs>(x => page.Appearing += x, x => page.Appearing -= x).Select(_ => true),
                     Observable.FromEventPattern<EventHandler, EventArgs>(x => page.Disappearing += x, x => page.Disappearing -= x).Select(_ => false));
             } else {
-                var cell = view as Cell;
+                var xfView = view as View;
 
-                if (cell != null) {
-                    ret = Observable
-                        .Merge(
-                            Observable.FromEventPattern<EventHandler, EventArgs>(x => cell.Appearing += x, x => cell.Appearing -= x).Select(_ => true),
-                            Observable.FromEventPattern<EventHandler, EventArgs>(x => cell.Disappearing += x, x => cell.Disappearing -= x).Select(_ => false));
+                if (xfView != null) {
+                    var propertyChanged = Observable.FromEventPattern<PropertyChangedEventHandler, PropertyChangedEventArgs>(
+                        x => xfView.PropertyChanged += x,
+                        x => xfView.PropertyChanged -= x);
+                    var parentChanged = propertyChanged
+                        .Where(x => x.EventArgs.PropertyName == "Parent")
+                        .Select(_ => Unit.Default);
+
+                    return parentChanged
+                        .StartWith(Unit.Default)
+                        .Select(_ => GetPageFor(xfView))
+                        .Select(x =>
+                            x == null ?
+                                Observable.Return(false) :
+                                Observable
+                                    .Merge(
+                                        Observable.FromEventPattern<EventHandler, EventArgs>(y => x.Appearing += y, y => x.Appearing -= y).Select(_ => true),
```

Comment:
Actually, reading into this a bit more, I don't think I'm going to make this change because it goes against all the advice I can find. Basically, if the event conforms to the normal pattern then the advice is to use `FromEventPattern`. Granted, it may be a tad slower, but I think it better to follow the official guidance.